File: src/api/MoRS.ManagementSystem.Application/DTOs/Room/UpdateRoomRequest.cs
using System.ComponentModel.DataAnnotations;

namespace MoRS.ManagementSystem.Application.DTOs.Room;

public class UpdateRoomRequest
{
    [Required]
    [StringLength(50, MinimumLength = 2)]
    public required string Name { get; set; }

    [Required]
    [StringLength(50, MinimumLength = 2)]
    public required string Type { get; set; }

    [Required]
    [RegularExpression("^#(?:[0-9a-fA-F]{3}){1,2}$")]
    public required string Color { get; set; }

    [Required]
    public bool IsActive { get; set; }
}

File: src/desktop/lib/models/room/update_room_request.dart
class UpdateRoomRequest {
  final String name;
  final String type;
  final String color;
  final bool isActive;

  UpdateRoomRequest({
    required this.name,
    required this.type,
    required this.color,
    required this.isActive,
  });

  factory UpdateRoomRequest.fromJson(Map<String, dynamic> json) {
    return UpdateRoomRequest(
      name: json['name'],
      type: json['type'],
      color: json['color'],
      isActive: json['isActive'],
    );
  }

  Map<String, dynamic> toJson() {
    return {'name': name, 'type': type, 'color': color, 'isActive': isActive};
  }
}
